pkg/status/collector/collector.go

```diff
@@ -0,0 +1,97 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package collector fetch information from different agent chekcs from expvar
+package collector
+
+import (
+	"encoding/json"
+	"expvar"
+)
+
+// UpdateStatus populates stats with collector information
+func UpdateStatus(stats map[string]interface{}) {
+	status := GetStatus()
+	for key, value := range status {
+		stats[key] = value
+	}
+}
+
+// GetStatus retrives collector information
+func GetStatus() map[string]interface{} {
+	stats := make(map[string]interface{})
+
+	runnerStatsJSON := []byte(expvar.Get("runner").String())
```

Comment:
For each of these vars, `runner`, `autoconfig`, `CheckScheduler`, `pyLoader`, and `pythonInit`, how about extracting out a helper function that handles the json unmarshaling. So instead these variables can be handled like this:

```
stats["runnerStats"] = unmarshalStatsVar("runner") // returns nil if var doesn't exist, or can't unmarshal
```

Reply:
I agree that we could extract the functionality of collecting the expvar. I would suggest delaying it until we have extracted all status sub packages, as we might find better helper functions or even sub packages to deal with expvars parsing.